utils/s3/client.hh

```diff
@@ -152,9 +152,20 @@ class client : public enable_shared_from_this<client> {
     future<> make_request(http::request req, reply_handler_ext handle, std::optional<http::reply::status_type> expected = std::nullopt, seastar::abort_source* = nullptr);
     future<> get_object_header(sstring object_name, http::experimental::client::reply_handler handler, seastar::abort_source* = nullptr);
 public:
-
-    client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag, std::unique_ptr<aws::retry_strategy> rs = nullptr);
+    client(std::string host,
+           endpoint_config_ptr cfg,
+           semaphore& mem,
+           global_factory gf,
+           private_tag,
+           std::unique_ptr<aws::retry_strategy> rs = nullptr,
+           std::unique_ptr<aws::aws_credentials_provider_chain> creds_provider_chain = nullptr);
     static shared_ptr<client> make(std::string endpoint, endpoint_config_ptr cfg, semaphore& memory, global_factory gf = {});
+    static shared_ptr<client> make(std::string endpoint,
+                                   endpoint_config_ptr cfg,
+                                   semaphore& mem,
+                                   global_factory gf,
+                                   std::unique_ptr<aws::retry_strategy> rs,
+                                   std::unique_ptr<aws::aws_credentials_provider_chain> cp_chain);
```

Comment:
nit: Why not just extra argument with default values to existing make()?

Reply:
nit-2: This change also allows specifying custom retry_strategy, which is not stated in patch commit